### DISCUSSION TITLE
CVE-2024-29187 fix broke setup scenario, where setup gets started by a service account

### DISCUSSION THREAD

**Discussion by rkieslinger** [wixtoolset/issues#8154] [WiX Development]
After updating from Wix 3.11.2.4516 to 3.14.1.8722, our setup fails if it gets started by a service account (Managed Service Account or Virtual Service Account).

![image](https://github.com/wixtoolset/issues/assets/20225125/f23b7c0f-e19e-4e94-87a2-b5f26fd6ff9a)

It seems the newly introduced check, if a process is elevated, always returns `true` if it is run by a service account. This results in the temp directory always getting created under `C:\Windows\Temp`, where then the service account without administrator privileges hasn't proper access.

I think the problem also exists in Wix 4 (and maybe 5) (seems to be the same code).

**Comment by bevanweiss** [comment]
In my very adhoc testing of Burn's handling, it did respect the TMP system variable to determine where to place the Temporary files.
So you could provide a folder that has permissions for your MSA / VSA..

**Comment by rkieslinger** [comment]
Thx for the information, but I don't really want to tamper with the TMP system variable. There's not really a good point in time when to temporarly change the TMP system variable and also this could have side effects.

IMO it's an obvious bug in an edge case.
